targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F437xG/TARGET_UBLOX_C030/min_battery_voltage.cpp

```diff
@@ -0,0 +1,49 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2013 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "min_battery_voltage.h"
+#include "battery_charger_i2c.h"
+
+/** Defining HAL_MspInit strong function
+ * in user defined file as described in documentation
+ * */
+
+void HAL_MspInit(void)
+{
+	set_minimum_battery_voltage();
+}
+
+void set_minimum_battery_voltage()
+{
+    char data_write[2] = {0};
+    char data_read;
+    BatteryChargerI2c i2c_object(I2C_SDA_B, I2C_SCL_B);
+
+    if (i2c_object.write_to_i2c(BQ24295_I2C_ADDRESS,&data_write[0] , 1)){
```

Comment:
I think your code here is correct, just change the comments as I indicate below.

Reply:
Alright